src/test/scala/mesosphere/marathon/core/task/TaskTest.scala

```diff
@@ -97,14 +97,14 @@ class TaskTest extends UnitTest with Inside {
       f.taskWithMultipleNetworksAndOneIp.status.networkInfo.effectiveIpAddress(f.appWithIpAddress).value should equal(f.ipString1)
     }
 
-    "effectiveIpAddress returns None if there is no ip" in {
+    "effectiveIpAddress returns agent hostname if there is no ip" in {
       val f = new Fixture
-      f.taskWithMultipleNetworkAndNoIp.status.networkInfo.effectiveIpAddress(f.appWithIpAddress) should be(None)
+      f.taskWithMultipleNetworkAndNoIp.status.networkInfo.effectiveIpAddress(f.appWithIpAddress) should be(Some("agent1.mesos"))
```

Comment:
Looks an important change; I think marathon was waiting for None in many parts of the code (and that's how it behave without CNI). Have you checked where effectiveIpAddress is used to make sure there is no other side effect than in the tests?

Reply:
There's always going to be an effective IP (if the CNI is off then it'll be the host's IP) so it's just to fix the tests